packages/core/src/html/NodeProcessor.js

```diff
@@ -315,6 +315,55 @@ class NodeProcessor {
     return node;
   }
 
+  static preprocessMarkdown(content) {
+    // Headings in Markdown are any line which is prefixed with a # symbol
+    // First, we break up the content into individual lines, and check which are prefixed with #
+    const lines = content.split('\n');
+    let amendedContent = '';
+    for (let i = 0; i < lines.length; i += 1) {
+      if (lines[i].charAt(0) === '#') {
+        // Next, we check if they fit the syntax of #s immediately followed by r/g/b then a space
+        const checker = lines[i].split(' ');
+        // Right now, we know that checker[0] starts with a #, so we just check the last character
+        const lastChar = checker[0].charAt(checker[0].length - 1);
+        if (lastChar === 'r' || lastChar === 'g' || lastChar === 'b') {
+          // Now that we've got the lines which match the syntax, we just
+          // replace the last character with <span> ... </span> and recreate the line
+          let span = '';
+          if (lastChar === 'r') {
+            span = '<span style="color: red;">';
+          } else if (lastChar === 'g') {
+            span = '<span style="color: green;">';
+          } else if (lastChar === 'b') {
+            span = '<span style="color: blue;">';
+          }
```

Comment:
Could we abstract this part out and instead use a map to represent this logic? It would make it easier to add new colors in the future and be more versatile overall.